runtime/doc/remote_ui.txt

```diff
@@ -0,0 +1,290 @@
+*remote_ui.txt*	Nvim
+
+
+		 NVIM REFERENCE MANUAL    by Thiago de Arruda
+
+
+Remote UI protocol				     *remote-ui*
+
+				      Type <M-]> to see the table of contents.
+
+==============================================================================
+1. Introduction						      *remote-ui-intro*
+
+GUIs can be implemented as external processes communicating with Nvim over the
+RPC API. The UI model consists of a terminal-like grid with a single,
+monospace font size. Some elements (UI "widgets") can be drawn separately from
+the grid ("externalized").
+
+                                                *remote-ui-options*
+After connecting to Nvim (usually a spawned, embedded instance) use the
+|nvim_ui_attach| API method to tell Nvim that your program wants to draw the
+Nvim screen on a grid of width × height cells. `options` must be
+a dictionary with these (optional) keys:
+	`rgb`			Controls what color format to use.
+				Set to true (default) to use 24-bit rgb
+				colors.
+				Set to false to use terminal color codes (at
+				most 256 different colors).
+	`ext_popupmenu`		Externalize the popupmenu. |ui-events-popupmenu|
+	`ext_tabline`		Externalize the tabline. |ui-events-tabline|
+	`ext_cmdline`		Externalize the cmdline. |ui-events-cmdline|
```

Comment:
I would prefer to keep the tags `ui-ext-foo`. "ext" is the token that means extensible/external widget.

Reply:
the point was to make it in line with `ui-events-grid` and `ui-events-mode`. Or we could reverse them and have `ui-events-global` and then `ui-events-grid`, the "grid" is some sense a widget, just the most important one. When we get multi-grid support I would rather have a "set current grid" event (prefixed in every update batch) than duplicating every single grid event (alternatively, add it as a field to every event).

Reply:
Maybe neither is needed: `ui-grid`, `ui-mode`, `ui-tabline`, ...

In all other contexts the `ui-` prefix is omitted, this means it's an editor feature.